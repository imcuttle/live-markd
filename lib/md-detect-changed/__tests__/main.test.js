/**
 * @file main
 * @author Cuttle Cong
 * @date 2018/10/27
 * @description
 *
 */
const detect = require('../')
const { fixture } = require('./helper')

const fs = require('fs')

function runDetect(name) {
  const oldMd = fs.readFileSync(fixture(name, 'old.md'))
  const newMd = fs.readFileSync(fixture(name, 'new.md'))
  return detect(oldMd, newMd)
}

describe('md-detect-changed', function() {
  it('should md-detect-changed works normal', () => {
    expect(runDetect('normal')).toContain('<p class="detected-updated">okkkk changed here</p>')
  })

  it('should md-detect-changed works diff-total', () => {
    expect(runDetect('diff-total')).toMatchInlineSnapshot(`
"<p class=\\"detected-updated\\">readme</p>
<p>fooo</p>"
`)
  })

  it('should md-detect-changed works diff-newline', () => {
    expect(runDetect('diff-newline')).toMatchInlineSnapshot(`
"<p>readme</p>
<p>fooo</p>"
`)
  })

  it('should md-detect-changed works diff-list', () => {
    expect(runDetect('diff-list')).toMatchInlineSnapshot(`
"<p>readme</p>
<ul>
<li class=\\"detected-updated\\">fooo</li>
</ul>
<p>asdasd</p>"
`)
  })

  it('should md-detect-changed works diff-new-list', () => {
    expect(runDetect('diff-new-list')).toMatchInlineSnapshot(`
"<h1 id=\\"readme\\"><a href=\\"#readme\\" class=\\"anchor\\"><svg aria-hidden=\\"true\\" class=\\"octicon octicon-link\\" height=\\"16\\" version=\\"1.1\\" view-box=\\"0 0 16 16\\" width=\\"16\\"><path d=\\"M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z\\"></path></svg></a>readme</h1>
<ul>
<li>fooo</li>
<li class=\\"detected-updated\\">new</li>
</ul>"
`)
  })

  it('should diff-rm-list', function() {
    expect(runDetect('diff-rm-list')).toMatchInlineSnapshot(`
"<h1 id=\\"readme\\"><a href=\\"#readme\\" class=\\"anchor\\"><svg aria-hidden=\\"true\\" class=\\"octicon octicon-link\\" height=\\"16\\" version=\\"1.1\\" view-box=\\"0 0 16 16\\" width=\\"16\\"><path d=\\"M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z\\"></path></svg></a>readme</h1>
<ul>
<li class=\\"detected-updated\\">fooo</li>
</ul>"
`)
  })

  it('should diff-rm-head', function() {
    expect(runDetect('diff-rm-head')).toMatchInlineSnapshot(`
"<h1 id=\\"readme\\"><a href=\\"#readme\\" class=\\"anchor\\"><svg aria-hidden=\\"true\\" class=\\"octicon octicon-link\\" height=\\"16\\" version=\\"1.1\\" view-box=\\"0 0 16 16\\" width=\\"16\\"><path d=\\"M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z\\"></path></svg></a>readme</h1>
<p>block block1</p>
<p class=\\"detected-updated\\">block block2</p>"
`)
  })

  it('should diff-code', function() {
    expect(runDetect('diff-code')).toMatchSnapshot()
  })
})
