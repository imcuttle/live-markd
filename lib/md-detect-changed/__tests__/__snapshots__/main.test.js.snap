// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`md-detect-changed should diff-code 1`] = `
"<h1 id=\\"ss\\"><a href=\\"#ss\\" class=\\"anchor\\"><svg aria-hidden=\\"true\\" class=\\"octicon octicon-link\\" height=\\"16\\" version=\\"1.1\\" view-box=\\"0 0 16 16\\" width=\\"16\\"><path d=\\"M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z\\"></path></svg></a>ss</h1>
<blockquote>
<p>cz adaptor for baidu BEFExs</p>
</blockquote>
<p class=\\"detected-updated\\">在项目根 <code>package.json</code> 中配置 new</p>
<pre><code class=\\"hljs language-text\\">...
  <span class=\\"hljs-string\\">\\"config\\"</span>: {
    <span class=\\"hljs-string\\">\\"commitizen\\"</span>: {
      <span class=\\"hljs-string\\">\\"path\\"</span>: <span class=\\"hljs-string\\">\\"cz-conventional-changelog-befe\\"</span>
    }
  }</code></pre>
<p>在项目中执行 <code>git cz</code></p>
<ul>
<li><code>git cz --retry</code>：跳过问答式交互，直接以上一次的提交 message 进行提交</li>
<li><code>git cz --read</code>: 使用 <code>git cz --read</code> 来使得上一次交互输入值成为默认值</li>
</ul>
<h2 id=\\"文档\\"><a href=\\"#%E6%96%87%E6%A1%A3\\" class=\\"anchor\\"><svg aria-hidden=\\"true\\" class=\\"octicon octicon-link\\" height=\\"16\\" version=\\"1.1\\" view-box=\\"0 0 16 16\\" width=\\"16\\"><path d=\\"M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z\\"></path></svg></a>文档</h2>
<ul>
<li><a href=\\"./docs/configuration.md\\">配置说明</a></li>
<li><a href=\\"./docs/github-usage.md\\">github 接入</a></li>
<li><a href=\\"./docs/gitlab-usage.md\\">gitlab 接入</a></li>
<li><a href=\\"./docs/icafe-usage.md\\">icafe 接入</a></li>
</ul>
<h2 id=\\"debugger\\"><a href=\\"#debugger\\" class=\\"anchor\\"><svg aria-hidden=\\"true\\" class=\\"octicon octicon-link\\" height=\\"16\\" version=\\"1.1\\" view-box=\\"0 0 16 16\\" width=\\"16\\"><path d=\\"M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z\\"></path></svg></a>Debugger</h2>
<pre><code class=\\"hljs language-bash\\"><span class=\\"hljs-comment\\"># 只调试 cz</span>
DEBUG=cz-conventional-changelog-befe git cz

<span class=\\"hljs-comment\\"># icafe-api 调试也开启</span>
DEBUG=cz-conventional-changelog-befe,icafe-api* git cz</code></pre>
<h2 id=\\"examples\\"><a href=\\"#examples\\" class=\\"anchor\\"><svg aria-hidden=\\"true\\" class=\\"octicon octicon-link\\" height=\\"16\\" version=\\"1.1\\" view-box=\\"0 0 16 16\\" width=\\"16\\"><path d=\\"M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z\\"></path></svg></a>Examples</h2>
<pre><code class=\\"hljs language-bash\\">git <span class=\\"hljs-built_in\\">clone</span> https://github.com/be-fe/cz-conventional-changelog-befe.git
<span class=\\"hljs-built_in\\">cd</span> cz-conventional-changelog-befe

npm i
npm run example
npm run example [<span class=\\"hljs-built_in\\">type</span>] <span class=\\"hljs-comment\\"># type 对应 examples 下的例子，如 github</span></code></pre>
<h2 id=\\"样例项目\\"><a href=\\"#%E6%A0%B7%E4%BE%8B%E9%A1%B9%E7%9B%AE\\" class=\\"anchor\\"><svg aria-hidden=\\"true\\" class=\\"octicon octicon-link\\" height=\\"16\\" version=\\"1.1\\" view-box=\\"0 0 16 16\\" width=\\"16\\"><path d=\\"M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z\\"></path></svg></a>样例项目</h2>
<p><a href=\\"http://icode.baidu.com/repos/baidu/personal-code/commit-demo\\">icode commit-demo</a></p>"
`;
